### PR TITLE
Build docs when touching docs/ and select directories in PRs

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

**Comment by foolip** [comment]
The docs building is broken on master now and we should catch it somehow. I used the path filtering to avoid showing it on all PRs, since the list of checks is already quite noisy.

**Comment by jugglinmike** [comment]
We're currently generating some documentation from files outside of `docs/` (and [we plan to do still more](https://github.com/web-platform-tests/wpt/pull/17189)), so it's possible that a pull request that doesn't touch `docs/` could still break the build.

I'm happy to merge this as-is since some visibility is better than no visibility, but I wanted to make sure you were aware of that, first.

**Comment by foolip** [comment]
Would also depending on tools/ and resources/ be enough right now? As long as the job doesn't trigger on most PRs that just update some tests I think it's fine if it over-triggers in directories that are mostly touched by core maintainers.

**Comment by jugglinmike** [comment]
Yeah, that sounds good to me!

File: .github/workflows/build-publish-documentation-website.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ on:
   push:
     branches:
       - master
+  pull_request:
+    paths:
+      - 'docs/*'
```
Comment (jugglinmike) [inline .github/workflows/build-publish-documentation-website.yml]: [It looks like we want two stars (`**`), not one.](https://help.github.com/en/articles/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)

Comment (foolip) [reply .github/workflows/build-publish-documentation-website.yml]: Thanks, fixed now!

**Review by jugglinmike — APPROVED** [review]

**Pull request merged**